gem/tests/test_views.py

```diff
@@ -637,15 +643,68 @@ def setUp(self):
         self.yourmind = self.mk_section(
             self.section_index, title='Your mind')
         self.client = Client(HTTP_HOST=self.main.get_site().hostname)
+        self.user = User.objects.create_user(
+            username='tester',
+            email='tester@example.com',
+            password='tester')
+
+    def test_reaction_question(self):
+        template_settings = deepcopy(settings.TEMPLATES)
+        template_settings[0]['DIRS'] = [
+            join(settings.PROJECT_ROOT, 'templates', 'chhaajaa')
+        ]
+
+        with self.settings(TEMPLATES=template_settings):
+            self.client.force_login(self.user)
+            promote_date = timezone.now() + timezone.timedelta(days=-1)
+            demote_date = timezone.now() + timezone.timedelta(days=1)
+            article = self.mk_article(
+                self.yourmind,
+                feature_as_hero_article=True,
+                promote_date=promote_date,
+                demote_date=demote_date
+            )
+            forms_index = FormsIndexPage.objects.filter().first()
+            form = create_molo_form_page(
+                forms_index, display_form_directly=True)
+
+            form_field = MoloFormField.objects.create(
+                page=form, sort_order=1, label='q1', field_type='checkboxes',
+                required=True, page_break=False,
+                admin_label='q1', skip_logic=None, choices='a,b,c',
+            )
+            ArticlePageForms.objects.create(
+                page=article, form=form)
+
+            res = self.client.get('/')
+            self.assertEqual(res.status_code, 200)
+            self.assertContains(res, article.title)
+            self.assertTrue(form.title in str(res.content))
+
+            self.assertTrue(form_field.label in str(res.content))
+
+            data = {
+                form_field.admin_label: 'b'
+            }
+            MoloFormSubmission.objects.create(
+                page=form, article_page=article,
+                user=self.user, form_data=data
+            )
+
+            res = self.client.get('/')
+            self.assertEqual(res.status_code, 200)
+            self.assertContains(res, article.title)
+            self.assertTrue(form.title in str(res.content))
+            self.assertFalse(form_field.label in str(res.content))
 
-    def test_highlights_correct_reaction(self):
+    def test_reaction_question_multi_submissions(self):
```

Comment:
I have a number of concerns with the changes to this test. I'll schedule a call with you to discuss it rather than us going back and forth here.